qiskit/quantum_info/operators/symplectic/stabilizer_table.py

```diff
@@ -652,6 +611,50 @@ def dot(self, other, qargs=None):
         """
         return super().dot(other, qargs=qargs)
 
+    def _compose(self, other, qargs=None, front=False):
+        if not isinstance(other, StabilizerTable):
+            other = StabilizerTable(other)
+        if qargs is None and other.num_qubits != self.num_qubits:
+            raise QiskitError("other StabilizerTable must be on the same number of qubits.")
+        if qargs and other.num_qubits != len(qargs):
+            raise QiskitError("Number of qubits in the other StabilizerTable does not match qargs.")
+
+        # Stack X and Z blocks for output size
+        x1, x2 = self._block_stack(self.X, other.X)
+        z1, z2 = self._block_stack(self.Z, other.Z)
+        phase1, phase2 = self._block_stack(self.phase, other.phase)
+
+        if qargs is not None:
+            ret_x, ret_z = x1.copy(), z1.copy()
+            x1 = x1[:, qargs]
+            z1 = z1[:, qargs]
+            ret_x[:, qargs] = x1 ^ x2
+            ret_z[:, qargs] = z1 ^ z2
+            pauli = np.hstack([ret_x, ret_z])
+        else:
+            pauli = np.hstack((x1 ^ x2, z1 ^ z2))
+
+        # We pick up a minus sign for products:
+        # Y.Y = -I, X.Y = -Z, Y.Z = -X, Z.X = -Y
+        if front:
+            minus = (x1 & z2 & (x2 | z1)) | (~x1 & x2 & z1 & ~z2)
+        else:
+            minus = (x2 & z1 & (x1 | z2)) | (~x2 & x1 & z2 & ~z1)
+        phase_shift = np.array(np.sum(minus, axis=1) % 2, dtype=np.bool)
+        phase = phase_shift ^ phase1 ^ phase2
+        return StabilizerTable(pauli, phase)
+
+    @classmethod
+    def _tensor(cls, a, b):
+        if not isinstance(a, StabilizerTable):
+            a = StabilizerTable(a)
+        if not isinstance(b, StabilizerTable):
+            b = StabilizerTable(b)
```

Comment:
This is from the signature of the `_tensor` class method. It gets called as `_tensor(cls, self, other)` in `tensor`, or `_tensor(cls, other, self)` in `expand`, so in practice only one of the args will possible be type changed, but `_tensor` doesn't know which one. I tried to leave all the type conversion in the class implementations rather than assume in the base class what type conversions would be happening.